woocommerce/woocommerce
# [Beta tester] Add tool for changing woocommerce.com base URL

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/team-ghidorah/issues/357

This PR adds a tool to the WooCommerce Beta Tester plugin that allows changing the base URL for the WooCommerce.com API. This is useful for testing remote specs changes without having to change the URL in the codebase.


![Screenshot 2024-09-19 at 16 26 10](https://github.com/user-attachments/assets/d8f7b09e-6083-47d5-8cd4-6450b66384fa)



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch locally.
2. cd to `plugins/woocommerce-beta-tester` and run `pnpm run build:zip` to create a zip file.
3. Use a site with Code Snippets plugin installed and activated.
4. Upload and activate woocommerce-beta-tester.zip
5. Add the following code snippet to the site:

```php
add_filter(
	'data_source_poller_data_sources',
	function ( $data_sources, $id ) {
		wc_get_logger()->debug( 'ID: ' . $id . ' Data Sources' . print_r( $data_sources, true ), array( 'source' => 'remote-specs' ) );
		return $data_sources;
	},
	10,
	2
);
```

6. Go to `wp-admin/tools.php?page=woocommerce-admin-test-helper` > Tools
7. Change woocommerce.com base URL to `https://staging.woocommerce.com` (a8c proxy needs to be enabled) or your local woocommerce.com URL
8. Run `wp transient delete --all` to delete the transients.
9. Go to `WooCommerce > Home`
10. Go to the Marketing page.
11. Go to WooCommerce > Status and check the logs for remote-specs source.
12. Confirm that the logs show the URL you set in step 7 for the remote specs data sources.


![Screenshot 2024-09-19 at 16 35 56](https://github.com/user-attachments/assets/d68b0b31-6aad-4465-82ad-de123235c116)




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
